todoapp/TODOapp/app/src/main/java/com/nearsoft/androidschool/todoapp/activities/detail/DetailActivity.java

```diff
@@ -5,50 +5,163 @@
 import android.support.design.widget.FloatingActionButton;
 import android.support.design.widget.Snackbar;
 import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.CardView;
 import android.view.View;
+import android.widget.CompoundButton;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.Switch;
 import android.widget.TextView;
 
 import com.nearsoft.androidschool.todoapp.R;
+import com.nearsoft.androidschool.todoapp.fragment.DatePickerFragment;
 import com.nearsoft.androidschool.todoapp.models.ToDoContent;
 
+import java.text.DateFormat;
+import java.util.Date;
+
 public class DetailActivity extends AppCompatActivity {
 
     public static final String EXTRA_TODO_KEY = "TODO";
 
-    private TextView titleTextView;
-    private TextView notesTextView;
+    private EditText titleEditTextView;
+    private EditText notesEditTextView;
+    private CardView dateCardView;
     private TextView dateTextView;
     private FloatingActionButton editFab;
+    private FloatingActionButton saveFab;
+    private Switch dateSwitch;
 
-    private ToDoContent todo;
+    private ToDoContent todoItem;
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        todo = getTodo();
+        setViews();
+        setSwitchListener();
+        populateToDoObject();
+        setSaveListener();
+        setEditListener();
+        setDateListener();
+    }
+
+    private void setViews() {
         setContentView(R.layout.activity_detail);
-        titleTextView = (TextView) findViewById(R.id.title);
-        notesTextView = (TextView) findViewById(R.id.notes);
-        dateTextView = (TextView) findViewById(R.id.date_text);
-        editFab = (FloatingActionButton) findViewById(R.id.fab);
+        titleEditTextView = (EditText) findViewById(R.id.title_field);
+        notesEditTextView = (EditText) findViewById(R.id.notes_field);
+        dateCardView = (CardView) findViewById(R.id.date_view);
+        editFab = (FloatingActionButton) findViewById(R.id.edit_fab);
+        saveFab = (FloatingActionButton) findViewById(R.id.save_fab);
+        dateTextView = (TextView) dateCardView.findViewById(R.id.date_text);
+        dateSwitch = (Switch) findViewById(R.id.switch_date);
+    }
+
+    private void setSaveListener() {
+        saveFab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    saveData();
+                    Snackbar.make(editFab, "Data Saved (Not saving in reality)", Snackbar.LENGTH_SHORT).show();
+                    buttonViewConfig(false);
+                } catch (AssertionError er) {
```

Comment:
do not catch AssertionErrors, an AssertionError is an error created by a developer.